test/integration/metrics/metrics_test.go

```diff
@@ -105,10 +105,7 @@ func TestAPIServerStorageMetrics(t *testing.T) {
 }
 
 func TestAPIServerMetrics(t *testing.T) {
-	// KUBE_APISERVER_SERVE_REMOVED_APIS_FOR_ONE_RELEASE allows for APIs pending removal to not block tests
-	t.Setenv("KUBE_APISERVER_SERVE_REMOVED_APIS_FOR_ONE_RELEASE", "true")
-
-	s := kubeapiservertesting.StartTestServerOrDie(t, nil, framework.DefaultTestServerFlags(), framework.SharedEtcd())
+	s := kubeapiservertesting.StartTestServerOrDie(t, &kubeapiservertesting.TestServerInstanceOptions{BinaryVersion: "1.33"}, framework.DefaultTestServerFlags(), framework.SharedEtcd())
```

Comment:
What does pinning BinaryVersion to an old minor do? This seems pretty weird and likely to confuse / break various layers. Are you expecting to leave this pinned to 1.33, or is this another thing that would need bumping at start of 1.34 dev?

Reply:
1.33 is pinned to the version `AdmissionregistrationV1beta1().ValidatingAdmissionPolicies` is deprecated. We have to pin the binary or emulation version so that we don't break the test once the version has moved across the border. Pinning emulation version would require we update this 3 versions later. Pinning the binary version would ensure this test works until the api is removed from the code base. 
We will work on a mock API to eventually replace this.

Reply:
We can use the `--emulated-version` flag like in https://github.com/kubernetes/kubernetes/pull/129165. I think that would be better